spec/compiler/parser/parser_spec.cr

```diff
@@ -927,6 +927,10 @@ describe "Parser" do
   assert_syntax_error "case {1, 2}; when {3}; 4; end", "wrong number of tuple elements (given 1, expected 2)", 1, 19
   assert_syntax_error "case 1; end", "unexpected token: end (expecting when or else)", 1, 9
 
+  it_parses "case! a\nwhen Int32\n1\nend", Case.new("a".call, [When.new(["Int32".path] of ASTNode, 1.int32)], check_exhaustiveness: true)
+  assert_syntax_error "case!\nwhen Int32\n1\nend", "unexpected token: when ('case!' must have a condition)", 2, 1
```

Comment:
I think that the error for this should be: `unexpected token: when ('case!' must have an expression)` (not necessarily a condition).